assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -6487,13 +6487,12 @@ private void prepareInlineForContentReview(String text, AssignmentSubmission sub
         //We will be replacing the inline submission's attachment
         //firstly, disconnect any existing attachments with AssignmentSubmission.PROP_INLINE_SUBMISSION set
         Set<String> attachments = submission.getAttachments();
-        for (String attachment : attachments) {
+        attachments.removeIf(attachment ->
+        {
             Reference reference = entityManager.newReference(attachment);
             ResourceProperties referenceProperties = reference.getProperties();
-            if ("true".equals(referenceProperties.getProperty(AssignmentConstants.PROP_INLINE_SUBMISSION))) {
-                attachments.remove(attachment);
```

Comment:
Note: this isn't directly related to the ticket, but I had to fix this bug that triggered ConcurrentModificationExceptions while testing resubmissions:
for (String attachment : attachments)
{
    ...
    attachments.remove(attachment); // <- CME here
}